org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/HoverInfoProvider.java

```diff
@@ -12,62 +12,75 @@
 
 import java.io.IOException;
 import java.io.Reader;
+import java.util.stream.Stream;
 
+import org.eclipse.core.runtime.CoreException;
 import org.eclipse.jdt.core.IBuffer;
 import org.eclipse.jdt.core.IJavaElement;
 import org.eclipse.jdt.core.ILocalVariable;
 import org.eclipse.jdt.core.IMember;
+import org.eclipse.jdt.core.IPackageFragment;
 import org.eclipse.jdt.core.ISourceRange;
-import org.eclipse.jdt.core.ISourceReference;
 import org.eclipse.jdt.core.ITypeParameter;
 import org.eclipse.jdt.core.ITypeRoot;
-import org.eclipse.jdt.core.JavaModelException;
-import org.eclipse.jdt.internal.core.util.Util;
-import org.eclipse.jface.text.BadLocationException;
-import org.eclipse.jface.text.DefaultLineTracker;
-import org.eclipse.jface.text.Document;
-import org.eclipse.jface.text.IDocument;
-import org.eclipse.jface.text.ILineTracker;
-import org.eclipse.jface.text.IRegion;
 import org.jboss.tools.vscode.java.internal.handlers.JsonRpcHelpers;
-
-import copied.org.eclipse.jdt.ui.JavadocContentAccess;
+import org.jboss.tools.vscode.java.internal.javadoc.JavadocContentAccess;
 
 public class HoverInfoProvider {
-	
+
 	private final ITypeRoot unit;
 	public HoverInfoProvider(ITypeRoot aUnit) {
 		this.unit = aUnit;
 	}
-	
+
 	public String computeHover(int line, int column) {
 		try {
 			IJavaElement[] elements = unit.codeSelect(JsonRpcHelpers.toOffset(unit.getBuffer(),line,column),0);
-			if(elements == null || elements.length != 1)
+			if(elements == null || elements.length == 0) {
 				return null;
-			IJavaElement curr= elements[0];
-//			return computeSourceHover(curr);
+			}
+			IJavaElement curr = null;
+			if (elements.length != 1) {
+				// they could be package fragments.
+				// We need to select the one that matches the package fragment of the current unit
+				IPackageFragment packageFragment = (IPackageFragment) unit.getParent();
+				IJavaElement found =
+						Stream
+						.of(elements)
+						.filter(e -> e.equals(packageFragment))
+						.findFirst()
+						.orElse(null);
+				if (found == null) {
```

Comment:
Because I think this is too early to check that. It can be a package from source as well if you import a type from another package. This is a case where it doesn't match the package from the current compilation unit.